src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,137 @@
+                size ? '' : 'tw-w-full',
+                disabled
+                    ? 'tw-border-line-x-strong'
+                    : 'hover:tw-border-line-x-strong focus-within:tw-border-line-xx-strong focus-within:hover:tw-border-line-xx-strong',
+                isFocusVisible && !passwordButtonIsFocusVisible && !clearButtonIsFocusVisible && FOCUS_STYLE,
+                status ? validationClassMap[status] : '',
+            ])}
+            data-test-id={dataTestId}
+        >
+            <input
+                aria-label="Password Input"
+                autoComplete={autocomplete ? 'on' : 'off'}
+                data-test-id={`${dataTestId}-input`}
+                disabled={disabled}
+                id={useMemoizedId(propId)}
+                onBlur={onBlur}
+                onFocus={onFocus}
+                onKeyDown={onKeyDown}
+                onChange={handleOnChange}
+                onClick={() => inputElementRef.current?.focus()}
+                placeholder={placeholder}
+                readOnly={readOnly}
+                ref={inputElementRef}
+                required={required}
+                size={size}
+                type={isObfuscated ? InputTypes.Password : InputTypes.Text}
+                value={value}
+                className={merge([
+                    'tw-w-full tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows tw-text-sm',
+                    disabled || readOnly
+                        ? 'tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40 hover:tw-cursor-not-allowed'
+                        : 'tw-text-black tw-placeholder-black-60 dark:tw-text-white',
+                ])}
+            />
+            <span className="tw-flex tw-justify-between tw-items-center t-max-w-sm">
+                <button
+                    {...passwordButtonFocusProps}
+                    aria-label={`${isObfuscated ? 'show' : 'hide'} text input`}
+                    onClick={() => setIsObfuscated(!isObfuscated)}
+                    data-test-id={`${dataTestId}-visibility-icon`}
+                    disabled={disabled}
+                    title="Toggle text visibility"
+                    type="button"
+                    className={merge([
+                        'tw-flex tw-items-center tw-justify-center tw-transition-colors tw-rounded tw-text-text-weak tw-p-1',
+                        disabled
+                            ? 'tw-cursor-default tw-text-black-40 hover:tw-cursor-not-allowed'
+                            : 'tw-text-black-60 hover:tw-text-black-100 hover:tw-rounded-sm hover:tw-bg-box-neutral hover:tw-text-box-neutral-inverse',
+                        passwordButtonIsFocusVisible && FOCUS_STYLE,
+                    ])}
+                >
+                    {isObfuscated ? <IconEye /> : <IconEyeOff />}
```

Comment:
If the size of the icon is not going to change, choose a sized one to import just like you did with IconCross.